ui/tools/generator.js

```diff
@@ -1,160 +1,12 @@
-		let type = `
-
-${component.nameTrim}Props = TypedDict('${component.nameTrim}Props', {`;
-		type += Object.entries(component.fields)
-			.map(([key, field]) => {
-				return `
-    "${key}": ${getPyType(field.type)}`;
-			})
-			.join(",");
-		type += `
-}, total=False)`;
-
-		type += `
-
-${component.nameTrim}EventHandlers = TypedDict('${component.nameTrim}EventHandlers', {`;
-		type += Object.entries(component.events || {})
-			.map(([key]) => {
-				return `
-    "${key}": Union[str, Callable]`;
-			})
-			.join(",");
-		type += `
-}, total=False)`;
-
-		const bindable = Object.entries(component.events || {}).filter(
-			([, ev]) => ev.bindable,
-		);
-		if (!bindable?.length) return type;
-
-		type += `
-
-${component.nameTrim}Bindings = TypedDict('${component.nameTrim}Bindings', {`;
-		type += bindable
-			.map(([key]) => {
-				return `
-    "${key}": str`;
-			})
-			.join(",");
-		type += `
-}, total=False)`;
-		return type;
-	});
-
-	return types.join("");
+async function generate() {
+	await generatorPythonUi.generate();
+	await generatorUiComponentJson.generate();
 }
 
-function generateClass() {
-	return `
-
-class StreamsyncUIManager(StreamsyncUI):
-    """The StreamsyncUIManager class is intended to include dynamically-
-    generated methods corresponding to UI components defined in the Vue
-    frontend during the build process.
-
-    This class serves as a bridge for programmatically interacting with the
-    frontend, allowing methods to adapt to changes in the UI components without
-    manual updates.
-    """
-    
-    # Hardcoded classes for proof-of-concept purposes
-  `;
-}
-
-function generateMethods(data) {
-	const methods = data.map((component) => {
-		const isBindable = Boolean(
-			Object.entries(component.events || {}).find(
-				([, ev]) => ev.bindable,
-			),
-		);
-		const bindParam = ` 
-            binding: Optional[${component.nameTrim}Bindings] = None,`;
-		const bindPass = `,
-            binding=binding`;
-		return `
-    @staticmethod
-    def ${component.nameTrim}(
-            content: ${component.nameTrim}Props = {},
-            *,
-            id: Optional[str] = None,
-            position: Optional[int] = None,
-            parentId: Optional[str] = None,
-            handlers: Optional[${component.nameTrim}EventHandlers] = None,
-            visible: Optional[Union[bool, str]] = None,${isBindable ? bindParam : ""}
-            ) -> Component:
-        """
-        ${component.description}
-        """
-        component = StreamsyncUI.${component.allowedChildrenTypes?.length ? "create_container_component" : "create_component"}(
-            '${component.type}',
-            content=content,
-            id=id,
-            position=position,
-            parentId=parentId,
-            handlers=handlers,
-            visible=visible${isBindable ? bindPass : ""})
-        return component
-    `;
-	});
-	return methods.join("");
+if (require.main === module) {
+	generate();
```

Comment:
The if statement here is unnecessary. Just call the function.

Reply:
I just reproduce a python pattern per habits. It allow to import any module as library without running the code inside.

Reply:
I know but in this file you don't have any exports so there is not point importing this file as a library.